### PR TITLE
Multi Filtering improvement to current filter plugin

### PR DESCRIPTION
**Pull request by islarky88**
add multi filters to a column. can ba and or or relation or both.

image

image

### CODE REVIEW

File: src/plugins/filter/filter.plugin.tsx [hunk 1]
```diff
@@ -185,7 +186,18 @@ export default class FilterPlugin extends BasePlugin {
 
   // called on internal component change
   private async onFilterChange(filterItem: FilterItem) {
-    this.filterByProps({ [filterItem.prop]: filterItem });
+    console.log('onFilterChange', filterItem);
```
Comment (m2a2x) [inline src/plugins/filter/filter.plugin.tsx:189]: why props was removed and why console log is left?

File: src/plugins/filter/filter.plugin.tsx [hunk 2]
```diff
@@ -185,7 +186,18 @@ export default class FilterPlugin extends BasePlugin {
...
+    // this.filterByProps({ [filterItem.prop]: filterItem });
+  }
+
+  private async onMultiFilterChange(filterItems: MultiFilterItem) {
+    console.log('onMultiFilterChange', filterItems);
```
Comment (m2a2x) [inline src/plugins/filter/filter.plugin.tsx:194]: remove please console.log

File: src/plugins/filter/filter.pop.tsx [hunk 3]
```diff
@@ -166,25 +215,83 @@ export class FilterPanel {
...
+    this.multiFilterChange.emit(this.filterItems);
+
     this.filterChange.emit({
       prop: this.changes.prop,
       type: this.changes.type,
       value: this.extraElement?.value?.trim(),
     });
-    this.changes = undefined;
+
+    // this.changes = undefined;
```
Comment (m2a2x) [inline src/plugins/filter/filter.pop.tsx:237]: why did you remove changes clear?

**Review by m2a2x — CHANGES_REQUESTED** [review]
need to remove console.logs

File: src/plugins/filter/filter.pop.tsx (hunk 3 above)
Comment (islarky88) [reply src/plugins/filter/filter.pop.tsx]: to not close the filter popup right away after adding a filter. very useful when adding multiple filters. just a better UX in my opinion

**Review by m2a2x — APPROVED** [review]
